tools/resourcedocsgen/cmd/docs/registry.go

```diff
@@ -132,89 +133,69 @@ func getSchemaFileURL(metadata pkg.PackageMeta) (string, error) {
 	return fmt.Sprintf("https://raw.githubusercontent.com/%s/%s/%s", repoSlug, metadata.Version, schemaFilePath), nil
 }
 
-func getRegistryPackagesPath(repoPath string) string {
-	return filepath.Join(repoPath, "themes", "default", "data", "registry", "packages")
-}
-
-func genResourceDocsForAllRegistryPackages(registryRepoPath, baseDocsOutDir, basePackageTreeJSONOutDir string) error {
-	registryPackagesPath := getRegistryPackagesPath(registryRepoPath)
-	metadataFiles, err := os.ReadDir(registryPackagesPath)
+func genResourceDocsForAllRegistryPackages(
+	provider svc.PackageMetadataProvider,
+	baseDocsOutDir, basePackageTreeJSONOutDir string,
+) error {
+	ctx := context.Background()
```

Comment:
Let's plumb the context through here.
```suggestion
func genResourceDocsForAllRegistryPackages(
	ctx conxtext.Context,
	provider svc.PackageMetadataProvider,
	baseDocsOutDir, basePackageTreeJSONOutDir string,
) error {
```